lib/rules/eqeqeq.js

```diff
@@ -20,13 +20,30 @@ module.exports = {
         schema: [
             {
                 enum: ["always", "smart", "allow-null"]
+            },
+            {
+                type: "object",
+                properties: {
+                    null: {
+                        enum: ["always", "never", "ignore"]
+                    }
+                },
+                additionalProperties: false
             }
```

Comment:
Just leaving a note here that I'm wondering if it's possible to use JSON Schema to enforce that the object option is only allowed if the first option is "always". If that's not possible, I have no problem with this going in as is.